RULES/cc/cc.go

```diff
@@ -33,7 +36,11 @@ func (obj ObjectFile) Build(ctx core.Context) {
 }
 
 func (obj ObjectFile) out() core.OutPath {
-	return obj.Src.WithExt("o")
+	defaultOut := obj.Src.WithExt("o")
+	if obj.OutDir == nil {
+		return defaultOut
+	}
+	return obj.OutDir.WithSuffix("/"+defaultOut.Relative())
```

Comment:
doesn't this duplicate the entire path? E.g.: `defaultOut` is `foo/bar/baz.o` and `obj.OutDir` is `foo/bar/mylib.a-OBJS`, won't we get `foo/bar/mylib.a-OBJS/foo/bar/baz.o`? Is that intentional? Is my understanding wrong?

Reply:
Yes it does, it's to avoid files with the same object name in different paths from conflicting

Reply:
It's very much an edge case, and two object files with the same name would probably also confuse something like a linker script, but I wanted to be robust.

Reply:
Make sense